### PR TITLE
feat: add redis integration to RfqmService

### PR DESCRIPTION
**Pull request by phil-ociraptor**
# Description

Add basic integration for Redis into RfqmService. Right now, simply logging. Will need to set up `REDIS_URI` in prod and staging environments to get this to work there.

Will also need a strategy for Ropsten.

# Testing Instructions

All tests are updated such that they pass

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/runners/http_rfqm_service_runner.ts [hunk 1]
```diff
@@ -146,6 +149,9 @@ export async function buildRfqmServiceAsync(connection: Connection, asWorker: bo
 
     const quoteServerClient = new QuoteServerClient(axiosInstance);
 
+    const redisClient = redis.createClient({ url: REDIS_URI });
```
Comment (opaolini) [inline src/runners/http_rfqm_service_runner.ts:152]: Will this throw an error when `REDIS_URI === undefined`?

Comment (phil-ociraptor) [reply src/runners/http_rfqm_service_runner.ts:152]: Currently playing around w/this variable, but will make sure to make it fail early if necessary!

File: src/constants.ts [hunk 2]
```diff
@@ -45,6 +46,7 @@ export const ONE_GWEI = new BigNumber(1000000000);
 // RFQM Service
 export const KEEP_ALIVE_TTL = ONE_MINUTE_MS * 5;
 export const RFQM_TRANSACTION_WATCHER_SLEEP_TIME_MS = ONE_SECOND_MS * 15;
+export const RFQM_NUM_BUCKETS = 100;
```
Comment (phil-ociraptor) [inline src/constants.ts:49]: @PirosB3 @rhinodavid - do you think this number is sufficiently large?

File: src/runners/http_rfqm_service_runner.ts (hunk 1 above)
Comment (phil-ociraptor) [reply src/runners/http_rfqm_service_runner.ts:152]: Alright! updated this with a sane default (localhost, which is the same default as we use for postgres)

**Comment by phil-ociraptor** [comment]
deploy staging

File: test/utils/deployment.ts [hunk 3]
```diff
@@ -152,6 +155,32 @@ async function confirmPostgresConnectivityAsync(maxTries: number = 5): Promise<v
         }
     }
 }
+async function confirmRedisConnectivityAsync(maxTries: number = 5): Promise<void> {
+    try {
+        await Promise.all([
+            // delay before retrying
+            new Promise<void>((resolve) => setTimeout(resolve, 2000)), // tslint:disable-line:custom-no-magic-numbers
```
Comment (rhinodavid) [inline test/utils/deployment.ts:162]: this rule drives me nuts

File: src/constants.ts (hunk 2 above)
Comment (phil-ociraptor) [reply src/constants.ts:49]: Yeah, the more I think about it, I think we should go higher to start, maybe like 1,000

File: test/utils/deployment.ts (hunk 3 above)
Comment (phil-ociraptor) [reply test/utils/deployment.ts:162]: yeah, not the biggest fan of it. Although it does do some good imo

**Review by rhinodavid — APPROVED** [review]
Onward

**Pull request merged**